src/mmrelay/db_utils.py

```
@@ -771,24 +836,128 @@ def _store(cursor: sqlite3.Cursor) -> None:
+
+    def _delete(cursor: sqlite3.Cursor) -> int:
+        """
+        Delete stale name rows using the bound table name and current ID set.
+        """
+        return _delete_stale_names_core(cursor, table_name, current_ids)
+
+    try:
+        deleted = manager.run_sync(_delete, write=True)
+        if deleted > 0:
+            logger.debug("Removed %d stale %s entries", deleted, name_type)
+        return deleted
```

Ruff (0.15.4)

[warning] Consider moving this statement to an `else` block

(TRY300)